src/instructlab/model/list.py

```diff
@@ -0,0 +1,132 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+import logging
+import os
+import time
+
+# Third Party
+from huggingface_hub import ModelCard
+from requests import exceptions as requests_exceptions
+import click
+import pandas
+
+# First Party
+from instructlab import clickext
+from instructlab.configuration import DEFAULTS
+from instructlab.model.backends.backends import is_model_gguf
+
+logger = logging.getLogger(__name__)
+
+
+@click.command(name="list")
+@clickext.display_params
+@click.option(
+    "--model-dirs",
+    help="Base directories where models are stored.",
+    multiple=True,
+    default=[DEFAULTS.MODELS_DIR],
+    show_default=True,
+)
+@click.option(
+    "--list-checkpoints",
+    type=click.BOOL,
+    default=False,
+    is_flag=True,
+    show_default=True,
+)
+def model_list(model_dirs: list[str], list_checkpoints: bool):
+    """lists models"""
+    data = []
+    for directory in model_dirs:
+        for entry in Path(directory).iterdir():
+            # if file, just tally the size. This must be a GGUF.
+            if entry.is_file() and is_model_gguf(entry):
+                # stat the gguf, add it to the table
+                stat = Path(entry.absolute()).stat(follow_symlinks=False)
+                f_size = stat.st_size
+                f_size, magnitude = convert_bytes_to_proper_mag(f_size)
+                # add to table
+                age = pandas.to_timedelta(time.time_ns() - stat.st_ctime_ns)
```

Comment:
ctime is probably the wrong thing. It's the time stamp of the last status change of the file's metadata (owner, permission, ...). mtime is the last file data modification time.

Reply:
ah cool, thanks!